generators/reference_chunking/noun_splitter/__init__.py

```diff
@@ -0,0 +1,28 @@
+from pydantic import BaseModel
+from extractors.util.spacy import SpacySingleton
+
+INPUT_EXAMPLE = {
+    "text": "My favorite noun is 'friend'.",
+    "spacy_model": "en_core_web_sm"
+}
+
+class NounSplitterModel(BaseModel):
+    text: str
+    spacy_model: str
+
+    class Config:
+        schema_extra = {"example": INPUT_EXAMPLE}
+
+
+def noun_splitter(req: NounSplitterModel):
+    """Creates embedding chunks based on the nouns in a text"""
+    nlp = SpacySingleton.get_nlp(req.spacy_model)
+    doc = nlp(req.text)
+
+    nouns_sents = []
+    for sent in doc.sents:
+        nouns = [token.text for token in sent if token.pos_ == "NOUN" and len(token.text) > 1]
+        if nouns:
+            nouns_sents.extend([" ".join(nouns[i:i+1]) for i in range(0, len(nouns), 1)])
+    return {"nouns": list(set(nouns_sents))}
```

Comment:
```suggestion
   nouns_sents = set()
    for sent in doc.sents:
        for token in sent:
            if token.pos_ == "NOUN" and len(token.text) > 1:
                nouns_sents.add(token.text)
                
    return {"nouns": list(nouns_sents)}
```
This version should be more efficient and more readable.
We could still go for list comprehension, but I think that makes it harder to read the function.

Just to be clear the scope of this brick is to extract the nouns from a text  and return them as list.
Related to chat, we had a similar method where we grouped the nouns in lists of three or four and returned a list of lists. But that's not what the brick should do right?